lib-injection/auto_inject.rb

```diff
@@ -0,0 +1,19 @@
+if ENV['DDTRACE_AUTOINJECT'] && !ENV['skip_autoinject']
+  if system 'skip_autoinject=true bundle show ddtrace'
+    puts 'ddtrace already installed...'
+  else
+    puts 'ddtrace is not installed... Perform auto-injection...'
+
+    if system 'skip_autoinject=true bundle add ddtrace'
+      puts 'ddtrace added to bundle...'
+    else
+      puts 'Something went wrong when adding ddtrace to bundle...'
```

Comment:
One common reason for failure of `bundle add` will be when the application uses Bundler in "deployment" mode: https://bundler.io/guides/deploying.html

"deployment" mode means the application maintainer wants to make sure the gems their application needs in order run are not tampered with.

We'll get an error like this in such scenarios when we run `bundle add`:
```
You are trying to install in deployment mode after changing your Gemfile.
Run `bundle install` elsewhere and add the updated Gemfile.lock to version control.

If this is a development machine, remove the /Gemfile freeze by running
`bundle config unset deployment`.
```

It's possible to circumvent this, as exemplified in the error message itself (e.g. `bundle config unset deployment`), but should we?